frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.test.ts

```diff
@@ -351,66 +354,68 @@ describe('sessionRecordingDataLogic', () => {
 
         it('loads each source, and on success reports recording viewed', async () => {
             await expectLogic(logic, () => {
-                logic.actions.loadRecordingSnapshots()
-                // loading the snapshots will trigger a loadRecordingSnapshotsSuccess
+                logic.actions.loadSnapshots()
+                // loading the snapshots will trigger a loadSnapshotsForSourceSuccess
                 // that will have the blob source
-                // that triggers loadRecordingSnapshots
+                // that triggers loadNextSnapshotSource
             }).toDispatchActions([
                 // the action we triggered
-                logic.actionCreators.loadRecordingSnapshots(),
-                'loadRecordingSnapshotsSuccess',
+                'loadSnapshots',
                 // the response to that triggers loading of the first item which is the blob source
                 (action) =>
-                    action.type === logic.actionTypes.loadRecordingSnapshots &&
+                    action.type === logic.actionTypes.loadSnapshotsForSource &&
                     action.payload.source?.source === 'blob',
-                'loadRecordingSnapshotsSuccess',
+                'loadSnapshotsForSourceSuccess',
                 // and then we report having viewed the recording
                 'reportViewed',
                 // the response to the success action triggers loading of the second item which is the realtime source
                 (action) =>
-                    action.type === logic.actionTypes.loadRecordingSnapshots &&
+                    action.type === logic.actionTypes.loadSnapshotsForSource &&
                     action.payload.source?.source === 'realtime',
-                'loadRecordingSnapshotsSuccess',
+                'loadSnapshotsForSourceSuccess',
                 // having loaded any real time data we start polling to check for more
-                'startRealTimePolling',
+                'pollRealtimeSnapshots',
+                // which in turn triggers another load
+                (action) =>
+                    action.type === logic.actionTypes.loadSnapshotsForSource &&
+                    action.payload.source?.source === 'realtime',
+                'loadSnapshotsForSourceSuccess',
             ])
         })
 
-        it('can start polling for snapshots', async () => {
+        it('polls up to a max threshold', async () => {
```

Comment:
🤯 I've just realised that the polling shouldn't trigger loading state in the logic - we don't want the loading affordances in the UI when we poll.

Totally separate to this, reading the code back made me realise